### PR TITLE
Use CI pre-commit action.

### LINKED ISSUE #104: Consider pre-commit-ci/lite
If I understand correctly, we could use https://github.com/pre-commit-ci/lite-action instead of the autocommit action to apply changes to PRs. This may work better than the current approach which creates a separate commit but doesn't run CI on it. 
We'd still have to run pre-commit ourselves.

### PR DESCRIPTION
**Pull request by YooSunYoung**
Fixes #104 
I updated the ``ci.yml`` of the ``copier_template`` workflow to use ``pre-commit`` action itself.
The ``template`` is not updated yet.

TODO: 
- [ ] Install pre-commit action in repositories (including ``copier_template``)
- [x] Add pre-commit action in ``template`` ci recipes

It was tested with ``essnmx``: https://github.com/scipp/essnmx/pull/12.

### CODE REVIEW

**Comment by YooSunYoung** [comment]
@SimonHeybrock 
Just need to check if `Install pre-commit action in repositories (including copier_template)` is done...!

**Comment by jl-wynen** [comment]
> What is the status here? @jl-wynen, you opened https://github.com/scipp/copier_template/issues/104 so maybe you are best as a reviewer

What @YooSunYoung said. As far as I'm concerned, we should switch to this action. I just installed the application in this repo. I'm happy to do it for all repos but I didn't want to do it without an explicit 'go ahead'.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,12 +18,14 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: '3.9'
-      - run: python -m pip install --upgrade pip
-      - run: python -m pip install -r requirements/ci.txt
-      - run: tox -e static
-      - uses: stefanzweifel/git-auto-commit-action@v5
+      - uses: pre-commit/action@v3.0.0
```
Comment (jl-wynen) [inline .github/workflows/ci.yml:21]: Not the latest version. Should we actually pin down to the patch version? We don't do this elsewhere.

Comment (YooSunYoung) [reply .github/workflows/ci.yml:21]: What do you mean by pin down...? 
And yeah, I see they 3.0.1 is released.

Comment (jl-wynen) [reply .github/workflows/ci.yml:21]: I meant 'down to' as a preposition. I.e., pin all parts of the version number. I think we often only pin the major and maybe minor version. But not the patch version.

Comment (YooSunYoung) [reply .github/workflows/ci.yml:21]: Ahh like that...! Okay, then I'll get rid of `.0.0`

Comment (jl-wynen) [reply .github/workflows/ci.yml]: Apparently this doesn't work. In that case, I'm fine with using the full version

Comment (YooSunYoung) [reply .github/workflows/ci.yml]: @jl-wynen Apparently `pre-commit` ci action doesn't work like that...

Comment (YooSunYoung) [reply .github/workflows/ci.yml]: https://github.com/scipp/copier_template/actions/runs/7841440116/job/21397772718

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**